### DISCUSSION TITLE
Overview of sites - light for status of backup

### DISCUSSION THREAD

**Discussion by Rolli1962** [akeeba/panopticon#280] [Ideas]
It would be nice to have a light in the overview of the sites about status of backup
In my case green if backup not older than 1 day, yellow if older than a week, red if older than a month.
And where I write this words - maybe other admins are thinking about other values. Make them changeable ;-)

Roland

**Comment by nikosdion** [comment]
If you have a hard line on the maximum time between backups – as you should! – why not set a backup schedule? Panopticon lets you run backups using its own CRON jobs as the backup trigger, i.e. it works with sites on hosts which have no support for CRON jobs, or sites on hosts with unreasonably limited CRON jobs. That was pretty much the point of having a remote backup API in the first place (if you remember, I had implemented something like this since 2007 with the XML-RPC integration in these old versions of Joomla).

If you have a use case where a warning makes more sense than automating the backup please do tell me. It might be something I have not / could not think of.

**Comment by Rolli1962** [comment]
I have set up backup schedule - before with an own script, now shifted to Panopticon.
It is for the Issue, that the backup was not successful (for whatever reason). Had it in the past, also after Upgrade to J5.0.0. So it is a hint that the admin has to go in the details and has to find out and fix the reason.

**Reply by nikosdion** [reply]
When the automatic backup fails, Panopticon sends you an email. Would it not make more sense to set up a mail filter for these emails?

**Comment by Rolli1962** [comment]
It is only an idea.
In my case all system-mails are stored in one folder, as unread. Should not been overseen. And my understanding with the next unsuccessful try a new mail will be send. Latest than it should be clear to do somewhat.
Think in the end only if the admin of panopticon has deleted the schedule for backup it could be come to a situation that no backup was done and even no email.

**Reply by nikosdion** [reply]
Would it make sense if there was a warning if there are failed backups taken by Panopticon?

The reason I am asking is that Panopticon can log its failures (that's already implemented as of this morning). Looking there is a cheap database query.

Checking the backup age, however, requires querying each site over the API about the list of backups. If we do that in the overview page and you have not opened Panopticon in the last few hours this could cause a timeout error without a cause that's obvious to the user.

**Reply by Rolli1962** [reply]
It is only an idea, not a request ;-)
Think it can work without a light.
Only - I thought when the readout take place of taken backups the youngest date of backup for this site can be stored in addition. In the site-view it has only to be compared the current date and those youngest backup date for the site.

**Reply by nikosdion** [reply]
You can't really have this feature be based on the last backup date and time without making sure you have the actually latest backup date and time.

For example, let's say that you cache this information for one hour and you have the trigger for showing the backup out of date warning at 24 hours. The previous backup was taken yesterday at 09:00. You visit Panopticon at 10:41 and see that the backup is out of date. You log into the site and take a backup at 10:52. Then you visit Panopticon again at 10:58. You would expect that Panopticon tells you the backup is up to date. In fact, it tells you that the backup is out of date because it still sees the data from your previous visit at 10:41.

The only way for this feature to work _reliably_ would be to poll the backup information every time you need to display it, including the overview. This API call takes 1 to 5 seconds on average. You display up to 50 sites by default, and let's say half of them use Akeeba Backup. This means that the page would take anywhere from 25 to 125 seconds to load. In the latter case you've probably hit a timeout, so you can't access Panopticon.

I understand what you are trying to do, I am telling you that the way you want this to be implemented is impractical and cannot happen. It's not like I haven't thought of it. This has been a feature in Akeeba Backup since the Joomla! 1.5 days.

The only way for this feature to work reliably would be to disable all backups EXCEPT those taken through Panopticon which is impractical for other reasons. Anything else means that you will never have actual data consistency between the site and Panopticon, only eventual data consistency. The vast majority of people can't understand that. So, instead of wasting my time explaining why this can't work the way people would expect it to work I'd rather not implement it at all. Fair?

**Reply by Rolli1962** [reply]
It is OK. Have not expected such a explanation. And many thanks !!!
Best - Roland

**Reply by nikosdion** [reply]
You're welcome! I always try to explain the reasoning behind refusing a feature. I try to implement everything that makes sense, as long as it doesn't cause a lot of issues 😉
